apps/drain-safe/src/App.tsx

```diff
@@ -0,0 +1,192 @@
+import React, { useCallback, useState, useEffect } from 'react';
+import { Button, Loader, Title, TextField, Table, Text } from '@gnosis.pm/safe-react-components';
+import { useSafeAppsSDK } from '@gnosis.pm/safe-apps-react-sdk';
+import web3Utils from 'web3-utils';
+import erc20 from './abis/erc20';
+import { fetchJson, encodeTxData } from './utils';
+import Container from './Container';
+import Icon from './Icon';
+import Flex from './Flex';
+
+interface Asset {
+  balance: string;
+  fiatBalance: string;
+  fiatConversion: string;
+  tokenInfo: {
+    address: string;
+    decimals: number;
+    logoUri: string;
+    name: string;
+    symbol: string;
+    type: string;
+  };
+}
+
+interface Balance {
+  fiatTotal: string;
+  items: Asset[];
+}
+
+const CURRENCY = 'USD';
+
+async function fetchSafeAssets(safeAddress: string, safeNetwork: string): Promise<Balance> {
+  const network = safeNetwork.toLowerCase() === 'mainnet' ? 'mainnet' : 'rinkeby';
+  const url = `https://safe-client.${network}.gnosis.io/v1/safes/${safeAddress}/balances/${CURRENCY}/?trusted=false&exclude_spam=true`;
+  const data = await fetchJson(url);
+  return data as Balance;
+}
+
+const App: React.FC = () => {
+  const { sdk, safe } = useSafeAppsSDK();
+  const [submitting, setSubmitting] = useState(false);
+  const [assets, setAssets] = useState<Asset[]>([]);
+  const [toAddress, setToAddress] = useState<string>('');
+  const [isFinished, setFinished] = useState<boolean>(false);
+
+  const fetchBalances = useCallback(async (): Promise<void> => {
+    // Fetch safe assets
+    try {
+      const data = await fetchSafeAssets(safe.safeAddress, safe.network);
+      setAssets(data.items);
+    } catch (err) {
+      console.error("Couldn't load assets", err);
+    }
+  }, [safe]);
+
+  const submitTx = useCallback(async () => {
+    setSubmitting(true);
+    setFinished(false);
+
```

Comment:
can we validate the address here? 
perhaps by calling `web3Utils.toChecksumAddress(toAddress)` and if it fails show an error in the input.

Reply:
Absolutely, good point 👌